src/drivers/intel/hda/hda-dma.c

```diff
@@ -665,8 +659,6 @@ static int hda_dma_set_config(struct dma_chan_data *channel,
 	if (channel->status == COMP_STATE_ACTIVE)
 		return 0;
 
-	irq_local_disable(flags);
```

Comment:
I think it would be great to have in this driver (or maybe in general for DMA drivers) comments which operations can be called from which interrupts, then it would be possible to understand where to disable interrupts and where not to. I understand the spinlock is taken only in `.channel_get()` and `.channel_put()` because only they scan the list of all channels on controllers, which can be accessed by different cores, whereas all other operations are only called on cores, using the channel.